### DISCUSSION TITLE
Lint acting strange

### DISCUSSION THREAD

**Discussion by Gregory1701** [codefori/vscode-ibmi#2188] [Q&A]
Hi all,
I'm trying really hard to like the RPG lint. But it continues to fall into my disfavor. I've had several odd behaviors, but every time I try to isolate that behavior into a simple program to share here, the linter works as expected.

First off, if I use Shift+Alt+F to format a pre-existing **Free source member sometimes it makes the code not compile. When I want to format the document, I will compile the source to ensure I'm working with a good working copy. Then Shift+Alt+F and then recompile the source. Most of the time, when it breaks the compile, it seems to be centered on SQL statements. But again, if I take that exact SQL statement and move it into a test program, the linter works fine. Now, as a fellow developer, I realize that without a sample to recreate the issue, it's tough to recreate/debug/fix. But if you PM me privately, I'd be happy to jump on a teams call and demo this for you.

This second example, again, if I isolate to a simple program, it works fine. But below is a print screen showing that the linter is claiming that the stand alone variables are not referenced; yet right below, they quite obviously are referenced/used.

![image](https://github.com/user-attachments/assets/b2a9ddc5-df72-46c5-90a8-a2feedfe5e0b)

**Comment by worksofliam** [comment]
@Gregory1701 we're going to need code snippets that can be copied/pasted so I can recreate these and create test cases where applicable. I know you offered a zoom but the linter is a little unique where we need replayable scenarios.

Thanks!

**Reply by Gregory1701** [reply]
Okay @worksofliam, I've got a source member that will recreate the issue; at least for me it does. I debated on how to get this to you as it's acting like there are some unseen characters that are in play here. For a first attempt, I've just copy/pasted the source into a text file (see attached). If this doesn't work, my next thought would be to copy the source member into a new source physical file, then do a SAVOBJ of the source physical file, then ftp it to my local, then send. Kind of a pain, but if that's what it takes to get this resolved, I'm happy to try that.

So in this source, there are 4 /copy directives, which you obviously don't have. But if I delete them, the 'No reference to definition' on line 29 goes away. Ctl+Z and it reappears. If I delete any of the stand alone variables or constants defined from line 13 to 23, the lint message on line 29 goes away. Ctl+Z to put them back and the lint message on 29 returns. Here's the really weird one; delete the blank line on line 31 and the line message on line 29 goes away, press enter to reinsert a new blank line, the lint message at line 29 returns. 

And you can see that the input parm on line 29 with the lint message is in the If statement on line 34.

[test_sqlrpgle.txt](https://github.com/user-attachments/files/16367331/test_sqlrpgle.txt)

**Reply by Gregory1701** [reply]
@worksofliam Were you able to recreate the issue with my example?

**Reply by worksofliam** [reply]
Honestly haven't had the time quite yet. I might be able to take a look tomorrow.

**Reply by worksofliam** [reply]
@Gregory1701 Can you also share the lint configuration with me which I can run against the source you provided?

**Reply by Gregory1701** [reply]
{
  "indent": 2,
  "PrototypeCheck": true,
  "NoOCCURS": true,
  "NoSELECTAll": true,
  "UppercaseConstants": false,
  "IncorrectVariableCase": true,
  "NoSQLJoins": false,
  "PrettyComments": true,
  "NoGlobalSubroutines": true,
  "NoUnreferenced": true,
  "SQLHostVarCheck": true,
  "BlankStructNamesCheck": true,
  "UselessOperationCheck": true,
  "NoExternalTo": [
    "QCMD",
    "QP2TERM",
    "QSH",
    "SYSTEM",
    "QCMDEXC"
  ]
}
